src/System.Private.ServiceModel/tools/scripts/OpenFirewallPorts.cmd

```diff
@@ -1,20 +1,37 @@
 @echo off 
 setlocal
 
-REM 
 echo Start Open all ports WCF test services use
-REM
 
-netsh advfirewall firewall add rule name="_WCF Test Server PortHttp" dir=in action=allow profile=any localport=8081 protocol=tcp
-netsh advfirewall firewall add rule name="_WCF Test Server PortHttps" dir=in action=allow profile=any localport=44285 protocol=tcp
-netsh advfirewall firewall add rule name="_WCF Test Server PortTcp" dir=in action=allow profile=any localport=809 protocol=tcp
-netsh advfirewall firewall add rule name="_WCF Test Server PortWebSocket" dir=in action=allow profile=any localport=8083 protocol=tcp
-netsh advfirewall firewall add rule name="_WCF Test Server PortWebSockets" dir=in action=allow profile=any localport=8084 protocol=tcp
+SET __EXITCODE=0
 
-REM we do not currently check the error level for the netsh advfirewall command. Will need to investigate the expected error
-REM code in all successful cases and add the check. Will also need to add remove the firewall rules script for clean up
-set __EXITCODE=%ERRORLEVEL%
+net session > NUL
+if NOT [%ERRORLEVEL%]==[0] (
+    set __EXITCODE=%_ErrorLevel%
+    echo Error: this script must be called in an elevated window
+	goto :done
+)
 
-endlocal
-exit /b %__EXITCODE% 
+call :AddRule "_WCF Test Server PortHttp" dir=in action=allow profile=any localport=8081 protocol=tcp"
 
+call :AddRule "_WCF Test Server PortHttps" dir=in action=allow profile=any localport=44285 protocol=tcp"
+
+call :AddRule "_WCF Test Server PortTcp" dir=in action=allow profile=any localport=809 protocol=tcp"
+
+call :AddRule "_WCF Test Server PortWebSocket" dir=in action=allow profile=any localport=8083 protocol=tcp"
+
+call :AddRule "_WCF Test Server PortWebSockets" dir=in action=allow profile=any localport=8084 protocol=tcp"
+
+goto :done
+
+:AddRule
+netsh advfirewall firewall add rule name=%1
```

Comment:
The only problem is goto :done does not exit from the function. It will still continue on a adding rule failure. I'll have to go back to my original way(without using a function).